### PR TITLE
chore: pull EE CLI features and docs into OSS [DET-3912]

### PR DESCRIPTION
**Pull request by ioga**
## Description

- move over EE diff for CLI and docs.
- add an error handling for 404 or missing data for EE-only calls.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: docs/reference/cluster-config.txt [hunk 1]
```diff
@@ -848,6 +848,43 @@ The master supports the following configuration settings:
                signed by a well-known CA; cannot be specified if
                ``skip_verify`` is enabled.
 
+-  ``scim``: (EE-only) Specifies whether the SCIM service is enabled and
```
Comment (stoksc) [inline docs/reference/cluster-config.txt:851]: I don't remember talking about pulling the docs in, but it makes a lot of sense, considering it's probably have the way anyone would figure out about EE.

File: harness/determined/cli/cli.py [hunk 2]
```diff
@@ -38,6 +40,8 @@
 from determined.deploy.cli import DEPLOY_CMD_NAME
 from determined.deploy.cli import args_description as deploy_args_description
 
+from .errors import MasterEERequired
```
Comment (stoksc) [inline harness/determined/cli/cli.py:43]: nit the name, maybe just EnterpriseOnly or something.

File: harness/determined/cli/sso.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+from determined.common import api
+from determined.common.declarative_argparse import Arg, Cmd
+
+from .errors import MasterEERequired
+
+CLI_REDIRECT_PORT = 49176
+
+
+def make_handler(master_url: str, close_cb: Callable[[int], None]) -> Any:
```
Comment (stoksc) [inline harness/determined/cli/sso.py:16]: this is my crowning achievement at determined

File: harness/determined/common/api/errors.py [hunk 4]
```diff
@@ -37,10 +37,15 @@ def __init__(self, response: requests.Response) -> None:
...
+class NotFoundException(APIException):
+    pass
+
+
 class UnauthenticatedException(BadRequestException):
     def __init__(self, username: str):
         super().__init__(
-            message="Unauthenticated: Please use 'det user login <username>' to log in"
+            message="Unauthenticated: Please use 'det user login <username>' for password login or"
+            " 'det auth login --provider=<provider>' to login through an SSO provider"
```
Comment (stoksc) [inline harness/determined/common/api/errors.py:48]: maybe put like "or for Enterprise users logging in with an SSO provider, use ... " so oss users dont misparse this

File: harness/determined/deploy/local/docker-compose.yaml [hunk 5]
```diff
@@ -25,6 +25,9 @@ services:
       DET_LOG_LEVEL: ${INTEGRATIONS_LOG_LEVEL:-info}
       DET_MASTER_HTTP_PORT: ${INTEGRATIONS_HOST_PORT:-8080}
       DET_DB_PASSWORD: ${DET_DB_PASSWORD}
+      DET_SCIM_ENABLED: ${DET_SCIM_ENABLED}
+      DET_SCIM_USERNAME: ${DET_SCIM_USERNAME}
+      DET_SCIM_PASSWORD: ${DET_SCIM_PASSWORD}
```
Comment (stoksc) [inline harness/determined/deploy/local/docker-compose.yaml:30]: probably can leave this and the det deploy stuff out for now. is it just me or does it looks like there's not even a way to supply them through the cli?

**Review by stoksc — COMMENTED** [review]
lgtm just minor comments

File: docs/reference/cluster-config.txt (hunk 1 above)
Comment (ioga) [reply docs/reference/cluster-config.txt:851]: I've cleared this with Angela, we're going to have unified docs.

File: harness/determined/deploy/local/docker-compose.yaml (hunk 5 above)
Comment (ioga) [reply harness/determined/deploy/local/docker-compose.yaml]: will do. there's no way to supply them here, and ee master doesn't consume these either.

File: docs/reference/cluster-config.txt (hunk 1 above)
Comment (stoksc) [reply docs/reference/cluster-config.txt:851]: sweet.

**Review by stoksc — APPROVED** [review]
lgtm

**Pull request merged**